services/apps/src/apps/clients/iplant_groups.clj

```diff
@@ -136,3 +138,25 @@
                     :form-params  (public-permission-update-body)
                     :content-type :json
                     :as           :json})))
+
+(defn- share-app*
+  "Shares an app with a user."
+  [app-id subject-id level]
+  (let [resource-name (grouper-app-resource-name app-id)
+        role-name     (grouper-user-group)]
+    (http/put (grouper-url "attributes" resource-name "permissions" "memberships" role-name subject-id level)
+              {:query-params {:user grouper-user}
+               :form-params  {:allowed true}
+               :content-type :json
+               :as           :json}))
+  nil)
+
+(defn share-app
+  "Shares an app with a user."
+  [app-id subject-id level]
+  (try+
+   (share-app* app-id subject-id level)
+   (catch :status {:keys [body]}
```

Comment:
Might use `clj-http-error?` from clojure-commons for this rather than just `:status`, it'll be a bit more specific.